src/estimation/include/iDynTree/Estimation/AttitudeEstimator.h

```diff
@@ -0,0 +1,142 @@
+     *
+     * The general procedure to use the estimators would be,
+     * - instantiate the filter,
+     * - set initial internal state
+     * - in a loop,
+     *     - update the filter with measurements
+     *     - propagate the states
+     *
+     * However, additional methods to set and get parameters for the filter might be available with respect to the filters.
+     */
+    class IAttitudeEstimator
+    {
+    public:
+        virtual ~IAttitudeEstimator();
+
+        /**
+         * @brief Update the filter with accelerometer and gyroscope measurements
+         *
+         * @param[in] linAccMeas left trivialized 3D vector of linear proper sensor acceleration measuements
+         * @param[in] gyroMeas left trivialized 3D vector of angular velocity measurements
+         *
+         * @note left trivialized angular velocity means the angular velocity of body frame B with respect to an inertial fram A, expressed in frame B
+         * @return true/false if successful/not
+         */
+        virtual bool updateFilterWithMeasurements(const iDynTree::LinearAccelerometerMeasurements& linAccMeas,
+                                          const iDynTree::GyroscopeMeasurements& gyroMeas) = 0;
+
+        /**
+         * @brief Update the filter with accelerometer, gyroscope and magnetometer measurements
+         *
+         * @param[in] linAccMeas left trivialized 3D vector of linear proper sensor acceleration measuements
+         * @param[in] gyroMeas left trivialized 3D vector of angular velocity measurements
+         * @param[in] magMeas left trivialized 3D vector of magnetometer measurements
+         *
+         * @note left trivialized angular velocity means the angular velocity of body frame B with respect to an inertial fram A, expressed in frame B
+         *
+         * @return true/false if successful/not
+         */
+        virtual bool updateFilterWithMeasurements(const iDynTree::LinearAccelerometerMeasurements& linAccMeas,
+                                                  const iDynTree::GyroscopeMeasurements& gyroMeas,
+                                                  const iDynTree::MagnetometerMeasurements& magMeas) = 0;
+
+        /**
+         * @brief Propagate the states and associated uncertainties through properly defined propagation functions
+         * The underlying implementation depends on the type of filter being implemented.
+         *
+         * @return true/false if successful/not
+         */
+        virtual bool propagateStates() = 0;
+
+        /**
+         * @brief Get orientation of the body with respect to inertial frame, in rotation matrix form
+         *
+         * @param[out] rot Rotation matrix
+         * @return true/false if successful/not
+         */
+        virtual bool getOrientationEstimateAsRotationMatrix(iDynTree::Rotation& rot) = 0;
+
+        /**
+         * @brief Get orientation of the body with respect to inertial frame, in quaternion form
```

Comment:
Each rotation matrix is associated with two quaternions. Which one is returned by this method?

Reply:
I think this question does not seem relevant in this context, since we do not obtain this by the mapping of SO(3) onto quaternion space. Rather, we obtain the quaternion directly through the quaternion dynamics as a function of the angular velocity.

Does this make sense ? Please correct me if I've understood wrong.

Reply:
I see. I suspect that depending on the specific filter and the trajectory of the system you could obtain one quaternion or another, so it could make sense not to promise anything here about which quaternion are you providing to the user.

Reply:
Yes. And I suppose it also mainly depends on the initial state as a quaternion that we feed to the filter.

Reply:
Definitely.